internal/manifests/collector/adapters/config_to_ports.go

```diff
@@ -217,7 +221,9 @@ func ConfigToMetricsPort(logger logr.Logger, config map[interface{}]interface{})
 	}
 
 	_, port, err := net.SplitHostPort(cOut.Service.Telemetry.Metrics.Address)
-	if err != nil {
+	if err != nil && strings.Contains(err.Error(), "missing port in address") {
```

Comment:
this mean we no longer propagate a bad error

Reply:
Does it make sense to write a warning in that case? e.g. "missing port in address, fall back to default %d".

Reply:
I recall this log line being a bit spammy, maybe it's better off as a debug?

Reply:
Also, we should make this into its own error type instead of using `string.Contains`, which is fragile.

Reply:
yeah very spammy, this removes that :)

Reply:
<img width="1046" alt="image" src="https://github.com/open-telemetry/opentelemetry-operator/assets/15680283/029261fa-c9a5-4b2b-b000-38f64f3950ff">
I tried for 30 mins with Jacob and others to do the type assertion on the error but the net package does not expose the error in a friendly way to make that possible. The Go compiler was complaining but I couldn't make it work and this was the closest I could get it. I am open to suggestions on how to fix this.

Reply:
I think the idiomatic way to do this is the following:

```go
	var addrErr *net.AddrError
	if err != nil {
		if errors.As(err, &addrErr) &&
			addrErr.Addr == cOut.Service.Telemetry.Metrics.Address &&
			addrErr.Err == "missing port in address" {
			return 8888, nil
		}
		return 0, err
	}
```
Somewhat confusingly, the second argument to errors.As should be a pointer to a pointer. But I don't think this actually makes things any more safe, so we can keep it as is.